### DISCUSSION TITLE
backup and restore?

### DISCUSSION THREAD

**Discussion by ralyodio** [btcpayserver/btcpayserver#2359] [Technical Support]
I need to move servers, what's the easiest way to backup and restore?

When I ran `./backup.sh --only-db` i get an error that container `pgdump_all` is not found.

**Comment by ralyodio** [comment]
I just don't want to have to setup my wallets again, so whatever minimal backup is fine.

**Reply by Zaxounette** [reply]
Hi chovy !

You tried to backup only the DB and got that error.

Did you try any of the other backup commands provided in the docs to backup everything, not just the DB ?
https://docs.btcpayserver.org/Docker/#how-can-i-back-up-my-btcpay-server

**Comment by ralyodio** [comment]
There’s no instructions on how to restore.

I already have a brand new instance running on my new server I just need the wallet info moved over.

**Reply by dennisreimann** [reply]
Good point, we should add docs for the restore process as well. I recently added a [PR that documents how to connect to the database](https://github.com/btcpayserver/btcpayserver-docker/pull/449/files) and this would be another good addition.

**Comment by dennisreimann** [comment]
Can you run `docker ps` and check if there is a container named `generated_postgres_1`? The backup script filters the containers using `"name=postgres_1"` to find the database container – the error message you get looks like this container cannot be found.

**Reply by ralyodio** [reply]
oh i stopped the service, that's probably why the backup script failed. Still just need the data. as I already have it running vanilla on my new server.

Need to know how to backup and restore like this.

**Comment by Kukks** [comment]
There is more to it than that. If you only restore the dB on a new
instance, Nbxplorer will not have the tx data and not even know to follow
the store wallets.

**Comment by AryanJ-NYC** [comment]
@chovy have you had any luck in BTCPay restoration?
